dolphin/alert_manager/trap_receiver.py

```diff
@@ -95,6 +97,25 @@ def _extract_oid_value(var_bind):
 
         return oid, val
 
+    @staticmethod
+    def _get_alert_source_by_host(source_ip):
+        """Gets alert source for given source ip address."""
+        filters = {'host': source_ip}
+        ctxt = context.get_admin_context()
+
+        alert_source = db.alert_source_get_all(ctxt, filters=filters)
+        if len(alert_source) == 0:
+            msg = "Alert source could not be found with host %s." \
+                  % source_ip
+            raise exception.AlertSourceNotFound(message=msg)
+
+        # This is to make sure unique host is configured each alert source
+        if len(alert_source) != 1:
```

Comment:
Are we expecting one to one mapping between sorce_ip and device. Probably that is not the case. In general One SNMP agent can manage mutiple devices and I hope source_ip in trap field is the Agent IP.

In case of VMAX , I believe source_ip is the IP of Solution Enabler where you configured snmp daemon ,  and which can manage mutiple VMAx arrays..
Please confirm..

Reply:
Currently we have no better way to identify and map alert source. Above scenario looks reasonable. Will continue to refine this. Added below issue for further enhancement
https://github.com/sodafoundation/SIM/issues/132